stackit/internal/services/observability/alertgroup/resource.go

```diff
@@ -0,0 +1,566 @@
+}
+
+// Schema defines the schema for the resource.
+func (a *alertGroupResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Observability alert group resource schema. Must have a `region` specified in the provider configuration.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: descriptions["id"],
+				Computed:    true,
+			},
+			"project_id": schema.StringAttribute{
+				Description: "STACKIT project ID to which the alert group is associated.",
+				Required:    true,
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"instance_id": schema.StringAttribute{
+				Description: descriptions["instance_id"],
+				Required:    true,
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Description: descriptions["name"],
+				Required:    true,
+				Validators: []validator.String{
+					validate.NoSeparator(),
+					stringvalidator.LengthBetween(1, 200),
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^[a-zA-Z0-9-]+$`),
+						"must match expression",
+					),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"interval": schema.StringAttribute{
+				Description: descriptions["interval"],
+				Optional:    true,
+				Validators: []validator.String{
+					validate.ValidDurationString(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"rules": schema.ListNestedAttribute{
+				Description: "Rules for the alert group",
+				Required:    true,
+				PlanModifiers: []planmodifier.List{
+					listplanmodifier.RequiresReplace(),
+				},
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"alert": schema.StringAttribute{
+							Description: descriptions["alert"],
+							Required:    true,
+							Validators: []validator.String{
+								stringvalidator.RegexMatches(
+									regexp.MustCompile(`^[a-zA-Z0-9-]+$`),
+									"must match expression",
+								),
+								stringvalidator.LengthBetween(1, 200),
+							},
+						},
+						"expression": schema.StringAttribute{
+							Description: descriptions["expression"],
+							Required:    true,
+							Validators: []validator.String{
+								stringvalidator.LengthBetween(1, 600),
+							},
+						},
+						"for": schema.StringAttribute{
+							Description: descriptions["for"],
+							Optional:    true,
+							Validators: []validator.String{
+								stringvalidator.LengthBetween(1, 600),
```

Comment:
according to the documentation, the string should have a length of 2..8
```suggestion
								stringvalidator.LengthBetween(2, 8),
```